lib/shippingwrapper.rb

```diff
@@ -0,0 +1,46 @@
+require 'active_shipping'
+
+class ShippingWrapper
+  attr_reader :fedex, :ups, :response
+  def initialize(fedex=nil, ups=nil, response=false)
+    if fedex.present? && ups.present?
+      @fedex = fedex
+      @ups = ups
+      @response = "Successful Request"
+     else
+      @fedex = fedex
+      @ups = ups
```

Comment:
Is it necessary to still set @fedex and @ups if it is an invalid request? If not, then you probably don't need to do it!